src/main/java/pizzashop/controller/KitchenGUIController.java

```diff
@@ -17,16 +17,19 @@
     @FXML
     public Button ready;
 
-    public static  ObservableList<String> order = FXCollections.observableArrayList();
+    protected static final ObservableList<String> order = FXCollections.observableArrayList();
     private Object selectedOrder;
     private Calendar now = Calendar.getInstance();
-    private String extractedTableNumberString = new String();
+    private String extractedTableNumberString = "";
     private int extractedTableNumberInteger;
     //thread for adding data to kitchenOrderList
-    public  Thread addOrders = new Thread(new Runnable() {
+    public final Thread addOrders = new Thread(new Runnable() {
         @Override
         public void run() {
-            while (true) {
+            int j=0;
+            boolean b=true;
+            while (b) {
+                j++;
                 Platform.runLater(new Runnable() {
                     @Override
                     public void run() {
```

Comment:
de ce ai facut asta?

Reply:
mi-a sugerat sonaru din nou, ma punea sa tratez cazu de break in afara if-ului din while, si daca il scriam intr-un else imi comenta ca break-ul apare in 2 locatii diferite si nu e ok

Reply:
ideea e ca in mare nu e echivalent algorimul cu ce era inainte. fa cum era inainte, pune e.printStackTrace() in loc de break si vezi daca mai "plange" sonarul. nu te chinui mai mult